webui/react/src/pages/FlatRuns/FlatRuns.tsx

```diff
@@ -0,0 +1,696 @@
+            break;
+          case V1LocationType.CUSTOMMETRIC:
+            dataPath = `summaryMetrics.${currentColumn.column}`;
+            break;
+          case V1LocationType.UNSPECIFIED:
+          default:
+            break;
+        }
+        switch (currentColumn.type) {
+          case V1ColumnType.NUMBER:
+            columnDefs[currentColumn.column] = defaultNumberColumn(
+              currentColumn.column,
+              currentColumn.displayName || currentColumn.column,
+              settings.columnWidths[currentColumn.column],
+              dataPath,
+            );
+            break;
+          case V1ColumnType.DATE:
+            columnDefs[currentColumn.column] = defaultDateColumn(
+              currentColumn.column,
+              currentColumn.displayName || currentColumn.column,
+              settings.columnWidths[currentColumn.column],
+              dataPath,
+            );
+            break;
+          case V1ColumnType.TEXT:
+          case V1ColumnType.UNSPECIFIED:
+          default:
+            columnDefs[currentColumn.column] = defaultTextColumn(
+              currentColumn.column,
+              currentColumn.displayName || currentColumn.column,
+              settings.columnWidths[currentColumn.column],
+              dataPath,
+            );
+        }
+        return columnDefs[currentColumn.column];
+      })
+      .flatMap((col) => (col ? [col] : []));
+    return gridColumns;
+  }, [
+    appTheme,
+    columnsIfLoaded,
+    isDarkMode,
+    projectColumns,
+    selectAll,
+    selection.rows,
+    settings.columnWidths,
+    settings.compare,
+    settings.pinnedColumnsCount,
+    users,
+  ]);
+
+  const onPageChange = useCallback(
+    (cPage: number, cPageSize: number) => {
+      updateSettings({ pageLimit: cPageSize });
+      // Pagination component is assuming starting index of 1.
+      setPage((prevPage) => {
+        if (cPage - 1 !== prevPage) {
+          setRuns(Array(cPageSize).fill(NotLoaded));
+        }
+        return cPage - 1;
+      });
+    },
+    [updateSettings],
+  );
+
+  const fetchRuns = useCallback(async (): Promise<void> => {
+    if (isLoadingSettings || Loadable.isNotLoaded(loadableFormset)) return;
+    try {
+      const tableOffset = Math.max((page - 0.5) * PAGE_SIZE, 0);
+      const response = await searchRuns(
+        {
+          //filter: filtersString,
+          limit: isPagedView ? settings.pageLimit : 2 * PAGE_SIZE,
+          offset: isPagedView ? page * settings.pageLimit : tableOffset,
+          projectId: project.id,
+          sort: sortString || undefined,
+        },
+        { signal: canceler.signal },
+      );
+      const loadedRuns = response.runs;
+
+      setRuns((prev) => {
+        if (isPagedView) {
+          return loadedRuns.map((run) => Loaded(run));
+        }
+
+        // Update the list with the fetched results.
+        return prev.toSpliced(
```

Comment:
i like this -- is it fully supported within the browsers we support?

Reply:
It's been [available in all major browsers since July 2023](https://caniuse.com/mdn-javascript_builtins_array_tospliced). I don't know if we have an official policy when it comes to using new browser features, but I think this should be fine.